lib/cchain/queryutil/staking.go

```diff
@@ -95,13 +95,20 @@ func DelegatorInflationRates(ctx context.Context, cprov cchain.Provider, delegat
 	return resp, false, nil
 }
 
-func allDelegators(ctx context.Context, cprov cchain.Provider) ([]sdk.AccAddress, error) {
+// DelegationBalance represents the total delegation balance of a delegator.
+type DelegationBalance struct {
+	DelegatorAddress sdk.AccAddress
+	Balance          sdk.Coin
+}
+
+// AllDelegations returns delegation balances of each unique delegator.
+func AllDelegations(ctx context.Context, cprov cchain.Provider) ([]DelegationBalance, error) {
 	vals, err := cprov.SDKValidators(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	uniq := make(map[string]sdk.AccAddress)
+	uniq := make(map[string]*DelegationBalance)
```

Comment:
avoid pointers if possible (prefer immutability over mutability). Just set the value again when updating.